include/handler.h

```diff
@@ -328,10 +329,29 @@ namespace detail {
 
 	// The current mechanism for hydrating the SYCL placeholder accessors inside Celerity accessors requires that the kernel functor
 	// capturing those accessors is copied at least once during submission (see also live_pass_device_handler::submit_to_sycl).
-	// As of SYCL 2020 kernel functors are passed as const references, so we have to do this manually here.
+	// As of SYCL 2020 kernel functors are passed as const references, so we explicitly copy or capture by reference in bind_kernel_with_*
+
+	template <typename Functor, int Dims>
+	auto bind_kernel_with_celerity_item(const Functor& kernel, const cl::sycl::range<Dims>& global_range) {
+		return [=](cl::sycl::item<Dims> s_item) { kernel(make_item<Dims>(s_item.get_id(), global_range)); };
+	}
+
 	template <typename Functor>
-	Functor copy_kernel(const Functor& kernel) {
-		return Functor{kernel};
+	[[deprecated("Support for kernels receiving cl::sycl::item<Dims> will be removed in the future, change parameter type to celerity::item<Dims>")]] //
+	auto
```

Comment:
For readability I'd prefer `Functor` here, underlining the fact that we are copying the kernel.

Reply:
Done.